packages/http-server/src/server.ts

```diff
@@ -58,7 +58,7 @@ const replyHandler = <LoaderInput>(
       const response = await prism.process({
         method: (req.method || 'get') as IHttpMethod,
         url: {
-          path: req.url || '/',
+          path: (req.url || '/').split('?')[0],
```

Comment:
Need to strip fragments, protocol, host, ect.. from the url so that is only the path.

Reply:
should add a test to verify what @pytlesk4 said.

also: 

- [ ] should not remove entire query string. we only need to remove our special query params (`__code`, `__dynamic`, `__contentType`, `__example`). Probably want to just create a const with those properties and use that wherever possible, so that things don't get missed if we add/change a new parameter in the future.
- [ ] add a test case to ensure those special query params are stripped out, and any others remain and are passed through to process

Reply:
actually @pytlesk4 that is part of the weirdness here. `url.path` is not the path, it needs to be the entire thing (proto and host included)

Reply:
@marbemac I think we should add properties on IHttpUrl for scheme, host and so on...

Also, surpriesd this isn't caught by typescript compiler, but the fastfitfy http query type is different than out query type: https://github.com/stoplightio/prism/blob/3ff4ff4e526ca20e287960b4b0a2235de9ff4f08/packages/http-server/src/server.ts#L62

Reply:
Is it? Looks like `type DefaultQuery = { [k: string]: any }` to me, which is compatible with our query typing.

Reply:
This is our query type: `[name: string]: string | string[]`

Reply:
> Need to strip fragments, protocol, host, ect.. from the url so that is only the path.

@pytlesk4 @marbemac 
I don't think req.url will contain all that information. Fastify server doesn't know what `host` it is running on. E.g. when you run a web server on `localhost:3000` and do `curl localhost:3000 /somepath` fastify will not receive the `localhost:3000` part.

HTTP protocol provides **only** paths, the `host` header is for virtual servers. Therefore an HTTP server should **only** care about a path, not the whole URL.

Therefore the idea of passing whole URL to fastify or prism does not seem to fit into HTTP protocol.